en/guide/follow_me.md

```diff
@@ -0,0 +1,75 @@
+
+
+## Set the Follow Configuration
+
+By default the vehicle will follow directly behind the target at a height and distance of 8 metres. 
+You can (optionally) call [set_config()](../api_reference/classdronecore_1_1_follow_me.md#classdronecore_1_1_follow_me_1a4b92c3a042911dd9bdb378c686458a34) at any time to specify a different height, follow distance, relative position (front left/right/centre or behind) and responsiveness to target movements. 
+
+The code fragment below shows how to set the configuration:
+```cpp
+// configure follow me behaviour
+FollowMe::Config config;
+config.min_height_m = 12.f;  // Minimum height
+config.follow_dist_m = 20.f;  // Follow distance
+config.responsiveness = 0.2f;  // Higher responsiveness
+config.follow_direction = FollowMe::Config::FollowDirection::FRONT;  //Follow from front-centre
+
+// Apply configuration
+bool configured = device.follow_me().set_config(config);
+if (configured) {
+    // handle config failure
+    std::cout << "Configuration failed" << std::endl;
+}
+```
+
+The [get_config()](../api_reference/classdronecore_1_1_follow_me.md#classdronecore_1_1_follow_me_1a054aebafe0839a1028f277285b769fe5) method is provided to get the current configuration:
+```cpp
+auto curr_config = device.follow_me().get_config();
+```
+
+## Following a Target
+
+To start and stop following a target, call [start()](../api_reference/classdronecore_1_1_follow_me.md#classdronecore_1_1_follow_me_1a694749d43d527f85584df25a49b05ccf) and [stop()](../api_reference/classdronecore_1_1_follow_me.md#classdronecore_1_1_follow_me_1a6394507b0fb96bceebe6efd17f0529ce), respectively.
+
+In between the calls, use [set_curr_target_location()](../api_reference/classdronecore_1_1_follow_me.md#classdronecore_1_1_follow_me_1afb8c24ed93421e904b0f528569c7699a) to set the target position for the vehicle to follow. 
+DroneCore automatically resends the last set position at the rate required by the autopilot, but this can be overridden by setting the location at another rate.
+
+> **Note** The API does not automatically get the position of the device running the DroneCore. The client app must fetch this using OS-specific methods and pass them to the `FollowMe` class.
+
+```cpp
+// Start following
+FollowMe::Result follow_me_result = device.follow_me().start();
+if (follow_me_result != FollowMe::Result::SUCCESS) {
+    // handle start failure (in this case print error)
+    std::cout << "Failed to start following:" << FollowMe::result_str(follow_me_result) << std::endl;
+}
+
+// ... Get target position from underlying platform or wherever and supply to vehicle 
+// Here we just show one point being set. Minimum values specified in the configuration are respected.
+follow_me.set_curr_target_location({ 47.39776569, 8.54553292, 9.0, 0.f, 0.f, 0.f });
+
+// Stop following
+follow_me_result = device.follow_me().stop();
+if (follow_me_result != FollowMe::Result::SUCCESS) {

```

Comment:
@shakthi-prashanth-m @julianoes Can this ever happen? ie if say we lost connection, won't the vehicle stop getting updates from DroneCore, so you'll stop following anyway?

What happens when you stop following? Does the mode change? What will the vehicle do - go to hold mode?

Reply:
@hamishwillee thats a good point. Thanks.
I simulated this scenario by pressing Ctrl-C in FollowMe mode. I monitored the mode for long time like after 2 mins or so, it remains in `AUTO.FOLLOW_TARGET`. I did see a [MAVLink command](http://mavlink.org/messages/common/#MAV_CMD_DO_FOLLOW) that controls how long a vehicle should be in FollowMe or FOLLOW_TARGET mode. But this command execution failed when I executed using MAVROS command-line.  I feel this should be part of FollowMe configuration; a PX4 parameter can be used to configure this value similar to other FollowMe configuration parameters. It is important to know how to use the above MAVLink command.

Reply:
@shakthi-prashanth-m . Hmmm, so the "intention" is that the vehicle go to hold mode when stop is called or after some timeout from having lost connection. I think you're probably right to provide this as a configuration value with a sensible default value (probably as part of existing `Config`) - because I suspect that otherwise this will end up being a semi-random value. What do you think @julianoes 

I'd be setting the parameter rather than this mavlink message. Did you try that?

Reply:
@hamishwillee , I would like to elaborate my explanation further:

When `FollowMe::stop()` is called, [we change](https://github.com/dronecore/DroneCore/blob/develop/plugins/follow_me/follow_me_impl.cpp#L167) the Vehicle mode to LOITER.
However, if we lost connection to Vehicle abruptly as you rightly pointed out, then the Vehicle remains at the previously published location in `AUTO.FOLLOW_TARGET` mode.
Like I said earlier, there is a MAVLink command to configure TTL seconds (duration in seconds for which Vehicle stays in FollowMe mode), and then switches to default hold mode. But I tried to use this command, it doesn't seem to work. I need to figure out how to use this command. Please let me know if you know it.

As I see there's no PX4 parameter to configure this TTL seconds value.
I didn't get your point about setting parameter, as there's none.

Reply:
@shakthi-prashanth-m Thanks for clarification.

I don't know if I'm looking in the right place but the [navigator code here](https://github.com/PX4/Firmware/blob/master/src/modules/navigator/follow_target.cpp#L344) seems to indicate that the vehicle will just wait forever (well, some failsafe) if no setpoint is received. In other words, I don't think the timeout has been implemented. I think we should raise a bug for that in PX4, and ideally fix it.

The fix should include a parameter that can be used to set the timeout - whether or not DroneCore actually calls an API to change that parameter is a different discussion.

Does that make sense?

Reply:
@hamishwillee  you're welcome.
I too vote for having PX4 parameter say `NAV_FT_TTL` (in secs) which decides how long the vehicle is in Follow Target mode (waiting for messages).

Reply:
Yes. Though I'd probably suggest we don't actually set it in the API (just document it).

Reply:
I assume you are pursuing that :-)

Reply:
@hamishwillee created [issue in PX4](https://github.com/PX4/Firmware/issues/8501) for adding this parameter.